### PR TITLE
fix: linting errors + migrate to new config system

### PR DESCRIPTION
**Pull request by pavanjoshi914**

### CODE REVIEW

File: frontend/.eslintrc.cjs [hunk 1]
```diff
@@ -24,6 +24,5 @@ module.exports = {
     "@typescript-eslint/no-unused-vars": ["warn", { args: "none" }],
     "no-console": ["error", { allow: ["info", "warn", "error"] }],
     "no-constant-binary-expression": "error",
-    curly: "error",
```
Comment (rolznz) [inline frontend/.eslintrc.cjs:27]: why remove? I added this to enforce conditionals have curly brackets

Comment (pavanjoshi914) [reply frontend/.eslintrc.cjs:27]: it triggers error even for single line statments which don't need to be wrapped usually. is it good practice?

Comment (rolznz) [reply frontend/.eslintrc.cjs:27]: I think so. It's an easy way to make a mistake

Comment (pavanjoshi914) [reply frontend/.eslintrc.cjs:27]: resolved

File: frontend/src/components/ui/navigation-menu.tsx [hunk 2]
```diff
@@ -116,13 +116,13 @@ NavigationMenuIndicator.displayName =
   NavigationMenuPrimitive.Indicator.displayName;
 
 export {
-  navigationMenuTriggerStyle,
```
Comment (rolznz) [inline frontend/src/components/ui/navigation-menu.tsx:119]: I don't think we should change the exports

**Review by rolznz — APPROVED** [review]
utACK

File: frontend/src/components/ui/button.tsx [hunk 3]
```diff
@@ -54,4 +54,5 @@ const Button = React.forwardRef<HTMLButtonElement, ButtonProps>(
 );
 Button.displayName = "Button";
 
+// eslint-disable-next-line react-refresh/only-export-components
```
Comment (rolznz) [inline frontend/src/components/ui/button.tsx:57]: Are these still needed after the config change?

**Pull request merged**